crypto/stateproof/coinGenerator.go

```diff
@@ -75,7 +76,7 @@ func makeCoinGenerator(choice *coinChoiceSeed) coinGenerator {
 	choice.version = VersionForCoinGenerator
 	rep := crypto.HashRep(choice)
 	shk := sha3.NewShake256()
-	shk.Write(rep)
+	shk.Write(rep) //nolint:errcheck
```

Comment:
I think we need to check/report error here

Reply:
I suspected that Write() on such an object can't fail, since it's not buffering anything, just doing some math.  That seems to be correct.  It will panic if a Write happens after a Read, but otherwise there is no error path in this Write() method.

Reply:
Writes to NewShake256 can't fail — I guess I'm already introducing bad conventions, you're supposed to really include an explanation like
//nolint:errcheck // Writes to ShakeHash can't return error, it only panics